src/permission/permission.h

```diff
@@ -0,0 +1,86 @@
+#ifndef SRC_PERMISSION_PERMISSION_H_
+#define SRC_PERMISSION_PERMISSION_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_options.h"
+#include "permission/child_process_permission.h"
+#include "permission/fs_permission.h"
+#include "permission/permission_node.h"
+#include "permission/worker_permission.h"
+#include "v8.h"
+
+#include <iostream>
+#include <map>
+
+namespace node {
+
+class Environment;
+
+namespace permission {
+
+#define THROW_IF_INSUFFICIENT_PERMISSIONS(env, perm_, resource_, ...)          \
+  do {                                                                         \
+    if (UNLIKELY(!(env)->permission()->is_granted(perm_, resource_))) {        \
+      node::permission::Permission::ThrowAccessDenied((env), perm_);           \
```

Comment:
Why are parentheses around `env` but not `perm_`? Also, this does run the risk of evaluating its arguments multiple times – Maybe this should be

```suggestion
    Environment* env_ = (env);                                                 \
    permission::Permission* perm = (perm_);;                                   \
    if (UNLIKELY(!env_->permission()->is_granted(perm, (resource_)))) {        \
      node::permission::Permission::ThrowAccessDenied(_env_, perm);            \
```

Reply:
It was caused by `make format-cpp`